enacts/enactstozarr.py

```diff
@@ -43,49 +120,148 @@ def regridding(data, resolution):
+        or as pathlib Paths.
+    var_name : str
+        name of the ENACTS variable in the nc files
+    time_res : str, optional
+        indicates the time resolution of the set of files.
+        Default is "daily" and other option is "dekadal"
+    zarr_resolution : real, optional
+        spatial resolution to regrid to.
+    chunks : int, tuple of int, "auto" or mapping of hashable to int, optional
+        Chunk sizes along each dimension X, Y and T.
     
+    Returns
+    -------
+    Xarray.Dataset containing variable `var_name` and coordinates X, Y and T
+
+    See Also
+    --------
+    xarray.open_mfdataset, set_up_dims, regridding, xarray.DataArray.chunk
+    """
     data = xr.open_mfdataset(
-        netcdf,
-        preprocess = partial(set_up_dims, time_res=time_res),
-        parallel=False
+        paths,
+        preprocess=partial(set_up_dims, time_res=time_res),
+        parallel=False,
     )[var_name]
     if zarr_resolution != None:
-        print("attempting regrid")    
+        print("attempting regrid")
         data = regridding(data, zarr_resolution)
+    return xr.Dataset().merge(data.chunk(chunks=chunks))
 
-    data = data.chunk(chunks=CONFIG['datasets'][time_res]['chunks'])
-    
-    if output_path == None:
-        zarr = f"{CONFIG['datasets'][time_res]['zarr_path']}{input_path}"
+
+def convert(
+    input_path,
+    output_path,
+    var_name,
+    time_res="daily",
+    zarr_resolution=None,
+    chunks={}
+):
+    """Converts a set of ENACTS files into zarr store.
+
+    Either create a new one or append an existing one
+
+    Parameters
+    ----------
+    input_path : str
+        path where the ENACTS nc files are
+    output_path : str
+        path where the zarr store is (to append to) or will be (to create).
+        To create, (last element of the) path is expected not to exist.
+        To append, path is expected to point to a zarr store.
+    var_name : str
+        name of the ENACTS variable in the nc files
+    time_res : str, optional
+        indicates the time resolution of the set of files.
+        Default is "daily" and other option is "dekadal"
+    zarr_resolution : real, optional
+        spatial resolution to regrid to.
+    chunks : int, tuple of int, "auto" or mapping of hashable to int, optional
+        Chunk sizes along each dimension X, Y and T.
+        
+    Returns
+    -------
+    output_path : where the zarr store has been written
+
+    See Also
+    --------
+    calc.read_zarr_data, filename2datetime, nc2xr, xarray.Dataset.to_zarr
+    """
+    print(f"converting files for: {time_res} {var_name}")
+    netcdf = list(sorted(Path(input_path).glob("*.nc")))
+    if Path(output_path).is_dir() :
+        current_zarr = calc.read_zarr_data(output_path)
+        last_T_zarr = current_zarr["T"][-1]
+        last_T_nc = filename2datetime(netcdf[-1], time_res=time_res)
+        if last_T_nc.strftime("%Y%m%d") < last_T_zarr.dt.strftime("%Y%m%d").values :
```

Comment:
Why the strftime? Doesn't this work?

```
if last_T_nc < last_T_zarr.values
```

Reply:
What you propose complains that it can't compare int with datetime.datetime
int is last_T_zarr.values
I had tried all sorts of things and I always run into comparing objects of different types...

Reply:
Hmm. Before posting that, I actually opened some random zarr dataset and tried `ds['T'][-1].values` and got a np.datetime64. Could you point me to the dataset you're testing this on so I can try to figure out what's different?

Reply:
these are the 2 objects we try to compare:

converting files for: daily precip
1983-05-31T00:00:00.000000000
<class 'numpy.datetime64'>
1983-05-31 00:00:00
<class 'datetime.datetime'>

Reply:
So `can't compare int with datetime.datetime` is a very unhelpful error message...

Two alternative suggestions:
- Replace `filename2datetime` with `filename2datetime64` which returns a `np.datetime64`.
- Convert the `datetime.datetime` to a `pd.Timestamp`, which is supposedly comparable with `datetime.datetime`

Reply:
the following works

np.datetime64(last_T_nc) < last_T_zarr.values

Reply:
That's better than my second suggestion. My first suggestion is arguably simpler.

Reply:
pushed...

sorry... I didn't figured that one out a dew days ago...